lib/inventory_sync/async/inventory_full_sync.rb

```diff
@@ -5,6 +5,11 @@ class InventoryFullSync < InventoryHostsSync
       set_callback :step, :around, :update_statuses_batch
 
       def plan(organization)
+        unless cloud_auth_available?
+          logger.debug('Cloud authentication is not available, skipping inventory hosts sync')
```

Comment:
From the UI we also disable to trigger this task if no token was found: https://github.com/theforeman/foreman_rh_cloud/blob/b7c22c6b2428f465cc5f06ac5ae5e7efa4aac577/webpack/ForemanInventoryUpload/Components/PageHeader/components/SyncButton/components/Modal.js#L24

though for rake tasks or some other ways to trigger the sync I guess it's better to return early, LGTM